usaspending_api/broker/management/commands/load_fsrs.py

```diff
@@ -28,123 +28,83 @@ def get_award_data(db_cursor, award_type, max_id, internal_ids=None):
-                    latest_transaction_id__isnull=False,
-                )
-                .distinct()
-                .order_by("-date_signed")
-                .first()
-            )
 
-            # We don't have a matching award for this subcontract, log a warning and continue to the next row
-            if not award:
-                msg = (
-                    "[Internal ID {}] Award not found for {},{},{},{}"
-                    + " (agency_id, referenced_idv_agency_iden, piid, parent_award_id)"
-                )
-                logger.warning(
-                    msg.format(
-                        row['internal_id'],
-                        row['contract_agency_code'],
-                        row['contract_idv_agency_code'],
-                        row['contract_number'],
-                        row['idv_reference_number'],
+        else:
+
+            # Find the award to attach this sub-grant to using the generated unique ID (unique_award_key):
+            if row['unique_award_key'] is not None:
+                all_awards = (
+                    Award.objects
+                    .filter(
+                        generated_unique_award_id=row['unique_award_key'],
+                        latest_transaction_id__isnull=False,
                     )
+                    .distinct()
+                    .order_by("-date_signed")
                 )
-                return None
-        else:
-            # Find the award to attach this sub-contract to. We perform this lookup by finding the Award containing
-            # a transaction with a matching fain
-            all_awards = (
-                Award.objects.annotate(modified_fain=Func(F('fain'), Value('-'), Value(''), function='replace'))
-                .filter(modified_fain=self.generate_unique_ids(row, award_type), latest_transaction_id__isnull=False)
-                .distinct()
-                .order_by("-date_signed")
-            )
 
-            if all_awards.count() > 1:
-                logger.warning("Multiple awards found with FAIN '{}'".format(row['fain']))
+                if all_awards.count() > 1:
+                    logger.warning(
+                        "Multiple awards found with generated_unique_award_id '{}'".format(row['unique_award_key']))
 
-            award = all_awards.first()
+                award = all_awards.first()
+
+        if not award:
+            msg = "[Internal ID {}] Award not found for unique_award_key '{}'"
+            logger.warning(msg.format(row['internal_id'], row['unique_award_key']))
+            return None
 
-            # We don't have a matching award for this subgrant, log a warning and continue to the next row
-            if not award:
-                msg = "[Internal ID {}] Award not found for FAIN '{}'"
-                logger.warning(msg.format(row['internal_id'], row['fain']))
-                return None
```

Comment:
Can you briefly explain what's happening in this part?

Reply:
I believe it's attempting to find USAspending awards for Broker subawards.  This is a supporting function.  The documentation for it is in `gather_shared_award_data` function.

> Creates a dictionary with internal IDs as keys that stores data for each award that's shared between all
> its subawards so we don't have to query the DB repeatedly
> Note: subawards without a matching award will still be processed, just without a matching id which may be matched later